src/http/device.ts

```diff
@@ -1374,20 +1374,57 @@ export class Lock extends Device {
+                }
+            }
+        } catch (error) {
+            this.log.error("Convert Error:", { property: property, value: value, error: error });
+        }
+        return super.convertRawPropertyValue(property, value);
+    }
+
     public processPushNotification(message: PushMessage, eventDurationSeconds: number): void {
         super.processPushNotification(message, eventDurationSeconds);
         if (message.type !== undefined && message.event_type !== undefined) {
             if (message.device_sn === this.getSerial()) {
                 try {
                     switch (message.event_type) {
-                        case LockPushEvent.LOCKED:
-                            this.updateProperty(PropertyName.DeviceLocked, { value: true, timestamp: message.event_time });
+                        case LockPushEvent.APP_LOCK:
+                        case LockPushEvent.AUTO_LOCK:
+                        case LockPushEvent.FINGER_LOCK:
+                        case LockPushEvent.KEYPAD_LOCK:
+                        case LockPushEvent.MANUAL_LOCK:
+                        case LockPushEvent.PW_LOCK:
+                            this.updateRawProperty(CommandType.CMD_DOORLOCK_GET_STATE, { value: "4", timestamp: convertTimestampMs(message.event_time) });
                             this.emit("locked", this, this.getPropertyValue(PropertyName.DeviceLocked).value as boolean);
                             break;
-                        case LockPushEvent.UNLOCKED:
-                            this.updateProperty(PropertyName.DeviceLocked, { value: false, timestamp: message.event_time });
+                        case LockPushEvent.APP_UNLOCK:
+                        case LockPushEvent.AUTO_UNLOCK:
+                        case LockPushEvent.FINGER_UNLOCK:
+                        case LockPushEvent.MANUAL_UNLOCK:
+                        case LockPushEvent.PW_UNLOCK:
+                            this.updateRawProperty(CommandType.CMD_DOORLOCK_GET_STATE, { value: "3", timestamp: convertTimestampMs(message.event_time) });
                             this.emit("locked", this, this.getPropertyValue(PropertyName.DeviceLocked).value as boolean);
                             break;
+                        case LockPushEvent.LOCK_MECHANICAL_ANOMALY:
+                        case LockPushEvent.MECHANICAL_ANOMALY:
+                        case LockPushEvent.VIOLENT_DESTRUCTION:
+                        case LockPushEvent.MULTIPLE_ERRORS:
+                            this.updateRawProperty(CommandType.CMD_DOORLOCK_GET_STATE, { value: "5", timestamp: convertTimestampMs(message.event_time) });
+                            break;
+                        case LockPushEvent.LOW_POWE:
+                            this.updateRawProperty(CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL, { value: "10", timestamp: convertTimestampMs(message.event_time) });
+                            break;
+                        case LockPushEvent.VERY_LOW_POWE:
```

Comment:
CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL corresponds to the battery value in %.

Are you sure that this message is always the same as 5% battery?

Reply:
Not sure but assuming. Looks like the lock is not sending back the battery level but a low or very low battery notification

Reply:
I would suggest that we comment out this part and as soon as we have a more precise idea of how we need to handle these events we will take up the topic again.